File: Seciovni.Web/wwwroot/js/HelperTypes/SearchResult.js
/**
 * Created by David on 2017-04-11.
 */

class SearchResult {
    constructor(){
        /**
         * The date the invoice was created
         * @type {Date}
         */
        this.createdDate = new Date();

        /**
         * The invoice's number
         * @type {number}
         */
        this.invoiceNumber = 0;

        /**
         * The buyer's full name
         * @type {string}
         */
        this.buyerName = "";

        /**
         * The sale's person's name
         * @type {string}
         */
        this.salesPerson = "";

        /**
         * The miscellaneous fees if any part of them were searched
         * @type {Array}
         */
        this.fees = [];

        /**
         * The payments if any part of them were searched
         * @type {Array}
         */
        this.payments = [];

        /**
         * The vehicles if any part of them were searched
         * @type {Array}
         */
        this.vehicles = [];

        /**
         * Any other fields that the user searched, where the key is the value of the dropdown
         * @type {object<string, string>}
         */
        this.otherFields = {};
    }
}

File: Seciovni.APIs/Contracts/SearchResult.cs
using Database.Tables;
using System;
using System.Collections.Generic;

namespace Seciovni.APIs.Contracts
{
    public class SearchResult
    {
        public DateTime CreatedDate { get; set; }
        public int InvoiceNumber { get; set; }
        public string BuyerName { get; set; }
        public string SalesPerson { get; set; }

        public IEnumerable<MiscellaneousFee> Fees { get; set; }
        public IEnumerable<Payment> Payments { get; set; }
        public IEnumerable<VehicleInfo> Vehicles { get; set; }
        public Dictionary<string, string> OtherFields { get; set; }
    }
}
